lib/Doctrine/Common/Collections/Expr/ClosureExpressionVisitor.php

```diff
@@ -37,6 +38,13 @@ class ClosureExpressionVisitor extends ExpressionVisitor
      */
     public static function getObjectFieldValue($object, string $field)
     {
+        if (strpos($field, '.') !== false) {
+            [$field, $subField] = explode('.', $field, 2);
```

Comment:
What should happen if there is more than one dot?

Reply:
I guess it would just won't work (just like it's not working without this PR)

Reply:
Can you describe "not working" more precisely? I want to know if the error message people are getting is clear enough right now.

Reply:
See issue https://github.com/doctrine/orm/issues/3783

Reply:
I think it looks good enough 👍